src/controllers/petsController.js

```diff
@@ -1,9 +1,38 @@
+const nodeGeocoder = require('node-geocoder');
+const geolib = require('geolib');
 const petsModel = require('../models/petsModel.js');
 const { inputValidation } = require("../utils/tools.js");
 const { ContentTypeError, PropNullorEmptyError, PropRequiredError } = require("../utils/errors.js");
 const { Logger } = require("../utils/log4js.js");
 const log = Logger();
 
+const options = {
+  provider: 'openstreetmap'
```

Comment:
It definitely is a bit slow like you said, but I would rather have that than spend money on this feature. I think this is a good option!